data/appsDownloadConfigs.yaml

```diff
@@ -0,0 +1,27 @@
+version: v1.0
+class: DownloadConfigs
+apps:
+  confetti:
+    Link: https://github.com/CBICA/precompiledBinaries/raw/master/windows/confetti.zip
+    Downloaded: 0
```

Comment:
Based on the Downloaded field here,  I assume this yaml file is modified to reflect when an app has been installed and then this is checked when we need to determine if an app is present. Is that correct?

If so:
Would it not be better to check for the presence of the expected app in the file system at run time? I know we had a discussion on where to download/install the apps (I think we decided on the user's home/.CaPTk/[version_number] directory), but I don't think we really accounted for the impact of having applications installed on a per-user basis. For example on a cluster with potentially 30 CaPTk users, we don't need 30 installs each of LIBRA, ITK-Snap, DeepMedic, etc. 
If we have a method to actually check for presence on the file system, then we could have it first check the system for a system-wide install, and then look in the user space (~/.CaPTk/[version_number]) only if we fail to find it. Presumably installing apps could work the same way, where we attempt first to place it alongside the CaPTk install and have the user space as a contingency.

The benefit of this would be that sysadmins could install CaPTk and the relevant apps as part of the installation process, but if they don't, the users can fall back to installing the apps they need without requiring root or system-wide access. I'll admit I'm not familiar enough with CaPTk's installation to know how feasible this is or if it would complicate things too much, but this would address one of the concerns that was raised today.

Reply:
Yes I had it there originally but switched to check the app at run time. That flag is not being used. 

For example, for libra, the check is done something like this
https://github.com/PhucNgo1711/CaPTk/blob/dc374ae38a2fb3a6d2ee93085d1c11d07208eb03/src/view/gui/fMainWindow.cpp#L5808